presto-docs/src/main/sphinx/connector/mongodb.rst

```diff
@@ -0,0 +1,242 @@
+
+This property is optional; the default is ``10000``.
+
+``mongodb.socket-timeout``
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The socket timeout in milliseconds. It is used for I/O socket read and write operations Socket.setSoTimeout(int)
+
+This property is optional; the default is ``0`` and means no timeout.
+
+``mongodb.socket-keep-alive``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This flag controls the socket keep alive feature that keeps a connection alive through firewalls Socket.setKeepAlive(boolean)
+
+This property is optional; the default is ``false``.
+
+``mongodb.read-preference``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The read preference to use for queries, map-reduce, aggregation, and count. The available values are PRIMARY, PRIMARY_PREFERRED, SECONDARY, SECONDARY_PREFERRED and NEAREST.
+
+This property is optional; the default is ``PRIMARY``.
+
+``mongodb.write-concern``
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The write concern to use. The available values are ACKNOWLEDGED, FSYNC_SAFE, FSYNCED, JOURNAL_SAFEY, JOURNALED, MAJORITY, NORMAL, REPLICA_ACKNOWLEDGED , REPLICAS_SAFE and UNACKNOWLEDGED.
+
+This property is optional; the default is ``ACKNOWLEDGED``.
+
+``mongodb.required-replica-set``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The required replica set name. With this option set, the MongoClient instance will
+
+#. Connect in replica set mode, and discover all members of the set based on the given servers
+#. Make sure that the set name reported by all members matches the required set name.
+#. Refuse to service any requests if any member of the seed list is not part of a replica set with the required name.
+
+This property is optional; no default value
+
+``mongodb.required-replica-set``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Limits the number of elements returned in one batch. A cursor typically fetches a batch of result objects and store them locally.
+If batchSize is 0, Driver's default will be used.
+If batchSize is positive, it represents the size of each batch of objects retrieved. It can be adjusted to optimize performance and limit data transfer.
+If batchSize is negative, it will limit of number objects returned, that fit within the max batch size limit (usually 4MB), and cursor will be closed. For example if batchSize is -10, then the server will return a maximum of 10 documents and as many as can fit in 4MB, then close the cursor.
+
+.. note::
+    Do not use a batch size of 1.
+
+This property is optional; the default is ``0``.
+
+Table Definition
+----------------
+
+MongoDB maintains table definitions on the special collection where ``mongodb.schema-collection`` configuration value specifies.
+
+.. note::
+    There's no way for the plugin to detect a collection is deleted.
+    You need to delete the entry by ``db.getCollection("_schema").remove( { table: deleted_table_name })``. Or please drop a collection by ``drop table table_name`` at Presto shell.
```

Comment:
User might wonder why a table is still shown at `show tables` after he have deleted a collection through his mongo shell. This is a node for this case.